src/Plugins/RpcServer/RpcServer.SmartContract.cs

```diff
@@ -212,81 +212,122 @@ private static Witness[] WitnessesFromJson(JArray _params)
+        /// <param name="useDiagnostic">Optional. Flag to enable diagnostic information.</param>
+        /// <returns>A JToken containing the result of the invocation.</returns>
+        [RpcMethodWithParams]
+        protected internal virtual JToken InvokeFunction(string scriptHash, string operation, ContractParameter[] args = null, Signer[] signers = null, bool useDiagnostic = false)
         {
-            UInt160 script_hash = Result.Ok_Or(() => UInt160.Parse(_params[0].AsString()), RpcError.InvalidParams.WithData($"Invalid script hash {nameof(script_hash)}"));
-            string operation = Result.Ok_Or(() => _params[1].AsString(), RpcError.InvalidParams);
-            ContractParameter[] args = _params.Count >= 3 ? ((JArray)_params[2]).Select(p => ContractParameter.FromJson((JObject)p)).ToArray() : System.Array.Empty<ContractParameter>();
-            Signer[] signers = _params.Count >= 4 ? SignersFromJson((JArray)_params[3], system.Settings) : null;
-            Witness[] witnesses = _params.Count >= 4 ? WitnessesFromJson((JArray)_params[3]) : null;
-            bool useDiagnostic = _params.Count >= 5 && _params[4].GetBoolean();
-
+            UInt160 contractHash = Result.Ok_Or(() => UInt160.Parse(scriptHash), RpcError.InvalidParams);
             byte[] script;
             using (ScriptBuilder sb = new())
             {
-                script = sb.EmitDynamicCall(script_hash, operation, args).ToArray();
+                script = sb.EmitDynamicCall(contractHash, operation, args ?? Array.Empty<ContractParameter>()).ToArray();
             }
-            return GetInvokeResult(script, signers, witnesses, useDiagnostic);
+            return GetInvokeResult(script, signers, [], useDiagnostic);
         }
 
-        [RpcMethod]
-        protected internal virtual JToken InvokeScript(JArray _params)
+        /// <summary>
+        /// Returns the result after passing a script through the VM.
+        /// </summary>
+        /// <remarks>
+        /// This method is to test your VM script as if they ran on the blockchain at that point in time.
+        /// This RPC call does not affect the blockchain in any way.
+        /// You must install the plugin RpcServer before you can invoke the method.
+        /// </remarks>
+        /// <param name="scriptBase64">A script runnable in the VM, encoded as Base64. e.g. "AQIDBAUGBwgJCgsMDQ4PEA=="</param>
+        /// <param name="signers">Optional. The list of contract signature accounts.</param>
+        /// <param name="witnesses">Optional. The list of witnesses for the transaction.</param>
+        /// <param name="useDiagnostic">Optional. Flag to enable diagnostic information.</param>
+        /// <returns>A JToken containing the result of the invocation.</returns>
+        [RpcMethodWithParams]
+        protected internal virtual JToken InvokeScript(string scriptBase64, Signer[] signers = null, Witness[] witnesses = null, bool useDiagnostic = false)
```

Comment:
It's an RPC protocol change, we can't change it so easily. Second parameter should be an array of signers with witnesses.

Reply:
@chenzhitong pease update the rpc document accordingly.

Reply:
@AnnaShaleva can you please check the update? Not really sure if we allow signer and witness at the same time, no document at all.

Reply:
> if we allow signer and witness at the same time

We do.